src/library_webgpu.js

```diff
@@ -41,14 +41,23 @@
     },
 
     makeU64ToNumber: function(lowName, highName) {
-      var ret = '('
+      var ret = '(';
       if (ASSERTIONS) {
         ret += 'assert(' + highName + ' < 0x200000), ';
       }
       ret += highName + ' * 0x100000000 + ' + lowName + ')\n'
       return ret;
     },
 
+    makeU64ToNumberWithSentinelAsUndefined: function(lowName, highName) {
+      var ret = '((' + highName + ' === 0xFFFFFFFF && ' + lowName + ' === 0xFFFFFFFF) ? undefined : (';
```

Comment:
You can use JS string template literals in the template code. e.g. `'((${highName} === 0xFFFFFFFF && ${lowName} === 0xFFFFFFFF) ? undefined : (`;

Reply:
Ah, I didn't realize we could use template literals in Emscripten. Is there documentation on what features we can use? (E.g. can we use arrow functions and const/let?)

Reply:
Not much in the way of documentation I'm afraid, but as rule you can use pretty much latest feature in the compiler itself (i.e. code the runs at build time) like this.     So you can use `const` and `let` and arrow functions (which we do in parseTools.js, jsifier.js, etc).

For generated code the situation is very different and you have to stick to ES5 today but I'm hoping to change that soon #11984

Reply:
Ahh, that makes sense. Looking forward to #11984!